### PR TITLE
series: add new exercise

### PR DESCRIPTION
**Pull request by filiptoskovic**

### CODE REVIEW

File: exercises/series/uSeriesTest.pas [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+implementation
+
+uses System.SysUtils;
+
+procedure TSeriesTest.CompareArrays(Array1, Array2: TArray<string>);
+var
+  i: integer;
+begin
+  Assert.AreEqual(Length(Array1), Length(Array2), ' - Arrays doesn''t have same length');
```
Comment (rpottsoh) [inline exercises/series/uSeriesTest.pas]: `Array lengths must be equal` reads a little better.

File: exercises/series/uSeriesTest.pas [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  for i := Low(Array1) to High(Array1) do
+    Assert.AreEqual(Array1[i], Array2[i], format(' - on [%d] place of array', [i]));
```
Comment (rpottsoh) [inline exercises/series/uSeriesTest.pas]: How about?:
```pascal
  Assert.AreEqual(Array1[i], Array2[i], format('Expecting element %d to = %s, Actual = %s',
    [i, Array1[i], Array2[i]]));
```

File: exercises/series/uSeriesTest.pas [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+end;
+
+procedure TSeriesTest.empty_series_is_invalid;
+var
+  Actual : TSlice;
+begin
+  Actual := TSlice.Create('');
+  Assert.WillRaiseWithMessage(procedure()
```
Comment (rpottsoh) [inline exercises/series/uSeriesTest.pas]: The `()` are not wrong after `procedure`, but they are also not needed.

File: exercises/series/uSeriesExample.pas [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+begin
+  if (FSerie = '') then
+    raise EArgumentException.Create('series cannot be empty');
+  case ALenght of
+    low(integer) .. -1 : raise EArgumentOutOfRangeException.Create('slice length cannot be negative');
+     0 : raise EArgumentOutOfRangeException.Create('slice length cannot be zero');
+    else
+      if (Length(FSerie) < ALenght) then
+        raise EArgumentOutOfRangeException.Create('slice length cannot be greater than series length');
+  end;
```
Comment (rpottsoh) [inline exercises/series/uSeriesExample.pas]: I think a series of if statements would be less clunky than this case.
```pascal
  if ALenght < 0 then
    raise EArgumentOutOfRangeException.Create('slice length cannot be negative');

  if aLenght = 0 then
    raise EArgumentOutOfRangeException.Create('slice length cannot be zero');

  if Length(FSerie) < ALenght then
    raise EArgumentOutOfRangeException.Create('slice length cannot be greater than series length');
```

File: exercises/series/uSeriesTest.pas (hunk 3 above)
Comment (filiptoskovic) [reply exercises/series/uSeriesTest.pas]: Yes, you are right. I have changed it.

File: exercises/series/uSeriesTest.pas [hunk 5]
```diff
@@ -142,7 +143,8 @@ procedure TSeriesTest.slice_length_cannot_be_negative;
   Actual : TSlice;
 begin
   Actual := TSlice.Create('123');
-  Assert.WillRaiseWithMessage(procedure()
+  Assert.WillRaiseWithMessage(
+    procedure
```
Comment (rpottsoh) [inline exercises/series/uSeriesTest.pas]: Nice.  Thanks

**Pull request merged**